BLOG.md

```diff
@@ -0,0 +1,363 @@
+# Telepresence with Docker Compose
+
+When it comes to developing an application, one of the potential friction points is the developer's machine. Several issues may arise:
+
+- Tight security policies may be in place.
+- The machine's operating system may differ significantly from the server's.
+- Each individual has their own setup and way of working.
+
+To address this challenge, platform engineers or developers themselves often resort to using a common solution that allows anyone to quickly start working on a project.
+
+One commonly seen tool for this purpose is Docker, and more specifically, Docker Compose.
+
+Both Compose and Telepresence assist developers in working on their applications, although they serve different purposes:
+
+- Compose provides a highly customizable way of running an entire application locally.
+- Telepresence enables running a local environment within a remote cluster.
+
+Let's explore how to harmonize and combine these two approaches.
+
+## A simple example
+
+Let's start with a simple example of infrastructure, and consider the following application :
+
+- A user-api micro service,
+- A contact-api micro service,
+- A Postgres database.
+
+The user API features an endpoint that retrieves user data, and in order to obtain the country information for each user, it makes additional requests to the contact API.
+
+These microservices store their data in a PostgreSQL database.
+
+Here is a schema summarizing the infrastructure:
+
+![Screenshot](images/compose-schema.png)
+
+And the equivalent docker-compose file.
+
+```yaml
+services:
+  contactapi:
+    build:
+      context: .
+      dockerfile: docker/contactapi/Dockerfile
+    environment:
+      DATASET: local
+      DB_HOST: postgres
+      DB_USERNAME: postgres
+      DB_PASSWORD: postgresql
+      PORT: 8081
+    depends_on:
+      - userapi
+    ports:
+      - 8081:8081
+
+  userapi:
+    build:
+      context: .
+      dockerfile: docker/userapi/Dockerfile
+    environment:
+      DATASET: local
+      DB_HOST: postgres
+      DB_USERNAME: postgres
+      DB_PASSWORD: postgresql
+      CONTACT_API_URL: "http://contactapi:8081"
+      PORT: 8080
+    depends_on:
+      - postgres
+    ports:
+      - 8080:8080
+
+  postgres:
+    image: postgres:15-bullseye
+    environment:
+      POSTGRES_PASSWORD: postgresql
+    ports:
+      - 5432:5432
+```
+
+You can run it locally with compose by running:
+
+```cli
+git clone https://github.com/datawire/telepresence-compose-demo
+cd telepresence-compose-demo
+docker compose up
+```
+
+And you should be able to see your service locally :
+
+```cli
+curl localhost:8080/users
+```
+
+As you can see, it only contains one test user, that's the local database.
+
+Press `Ctrl + C` to shutdown the Compose run.
+
+## Going Kubernetes
+
+Now let's say that you have an equivalent Kubernetes infrastructure.
+
+You can find its definition in the repository used above, and create it with:
```

Comment:
```suggestion
You can find the configuration at: <insert link to files here>, and create it with:
```

Reply:
Since this is a blogpost we want to be as direct as possible so we should include links instead of saying "found somewhere in what we listed somewhere else"